File: apps/push-service/src/main/java/com/example/chat/push/infrastructure/sender/TeamsPushSender.java
package com.example.chat.push.infrastructure.sender;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import java.util.List;
import java.util.Map;

/**
 * Microsoft Teams 푸시 발송자
 */
@Component
@RequiredArgsConstructor
@Slf4j
public class TeamsPushSender implements PushSender {
    private final RestTemplate restTemplate;

    @Override
    public boolean support(String pushType) {
        return "TEAMS".equalsIgnoreCase(pushType);
    }

    @Override
    public void send(String targetUserId, String title, String content) {
        String webhookUrl = "https://outlook.office.com/webhook/...";

        Map<String, Object> payload = Map.of(
                "type", "message",
                "attachments", List.of(Map.of(
                        "contentType", "application/vnd.microsoft.card.adaptive",
                        "content", Map.of(
                                "type", "AdaptiveCard",
                                "body", List.of(
                                        Map.of("type", "TextBlock", "text", title, "weight", "bolder", "size", "medium"),
                                        Map.of("type", "TextBlock", "text", content, "wrap", true)
                                ),
                                "$schema", "http://adaptivecards.io/schemas/adaptive-card.json",
                                "version", "1.0"
                        )
                ))
        );

        log.info("Sending Teams push to user {}", targetUserId);
        // restTemplate.postForEntity(webhookUrl, payload, String.class);
        log.info("Teams payload: {}", payload);
    }
}

File: apps/push-service/src/main/kotlin/com/example/chat/push/infrastructure/sender/TeamsPushSender.kt

package com.example.chat.push.infrastructure.sender

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

private val logger = KotlinLogging.logger {}

@Component
class TeamsPushSender(
    private val restTemplate: RestTemplate
) : PushSender {
    override fun support(pushType: String): Boolean = pushType.uppercase() == "TEAMS"

    override fun send(
        targetUserId: String,
        title: String,
        content: String
    ) {
        // In a real system, you'd fetch the Microsoft Teams Webhook URL
        val webhookUrl = "https://outlook.office.com/webhook/..."

        val payload = mapOf(
            "type" to "message",
            "attachments" to listOf(
                mapOf(
                    "contentType" to "application/vnd.microsoft.card.adaptive",
                    "content" to mapOf(
                        "type" to "AdaptiveCard",
                        "body" to listOf(
                            mapOf("type" to "TextBlock", "text" to title, "weight" to "bolder", "size" to "medium"),
                            mapOf("type" to "TextBlock", "text" to content, "wrap" to true)
                        ),
                        "\$schema" to "http://adaptivecards.io/schemas/adaptive-card.json",
                        "version" to "1.0"
                    )
                )
            )
        )

        logger.info { "Sending Teams push to user $targetUserId" }
        // Simulating the call
        // restTemplate.postForEntity(webhookUrl, payload, String::class.java)
        
        logger.info { "Teams payload: $payload" }
    }
}
